internal/services/fluidrelay/fluid_relay_servers_resource.go

```diff
@@ -312,3 +353,39 @@ func (s Server) Delete() sdk.ResourceFunc {
 func (s Server) IDValidationFunc() pluginsdk.SchemaValidateFunc {
 	return fluidrelayservers.ValidateFluidRelayServerID
 }
+
+func expandFluidRelayServerCustomerManagedKey(input []CustomerManagedKey) *fluidrelayservers.EncryptionProperties {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0]
+	encryption := &fluidrelayservers.EncryptionProperties{
+		CustomerManagedKeyEncryption: &fluidrelayservers.CustomerManagedKeyEncryptionProperties{
+			KeyEncryptionKeyUrl: pointer.To(v.KeyVaultKeyID),
+			KeyEncryptionKeyIdentity: &fluidrelayservers.CustomerManagedKeyEncryptionPropertiesKeyEncryptionKeyIdentity{
+				IdentityType:                   pointer.To(fluidrelayservers.CmkIdentityTypeUserAssigned),
+				UserAssignedIdentityResourceId: pointer.To(v.UserAssignedIdentityId),
+			},
+		},
+	}
+
+	return encryption
+}
+
+func flattenFluidRelayServerCustomerManagedKey(input *fluidrelayservers.EncryptionProperties) []CustomerManagedKey {
+	if input == nil || input.CustomerManagedKeyEncryption == nil {
+		return []CustomerManagedKey{}
+	}
+
+	customerManagedKey := CustomerManagedKey{}
+
+	if input.CustomerManagedKeyEncryption.KeyEncryptionKeyUrl != nil {
+		customerManagedKey.KeyVaultKeyID = pointer.From(input.CustomerManagedKeyEncryption.KeyEncryptionKeyUrl)
+	}
+	if input.CustomerManagedKeyEncryption.KeyEncryptionKeyIdentity != nil && input.CustomerManagedKeyEncryption.KeyEncryptionKeyIdentity.UserAssignedIdentityResourceId != nil {
+		customerManagedKey.UserAssignedIdentityId = pointer.From(input.CustomerManagedKeyEncryption.KeyEncryptionKeyIdentity.UserAssignedIdentityResourceId)
```

Comment:
We might want to parse the user assigned identity that we're getting back from the API here so that we can catch any casing inconsistencies
```suggestion
		if v := pointer.From(input.CustomerManagedKeyEncryption.KeyEncryptionKeyIdentity.UserAssignedIdentityResourceId); v != "" {
			id, err := commonids.ParseUserAssignedIdentity(v)
			if err != nil {
				return err
			}
			customerManagedKey.UserAssignedIdentityId = id.ID()
		}
```

Reply:
Good shout! Done!